src/libse/AudioToText/AudioToTextPostProcessor.cs

```diff
@@ -235,45 +252,33 @@ public Subtitle AddPeriods(Subtitle inputSubtitle, string language)
             {
                 var paragraph = subtitle.Paragraphs[index];
                 var next = subtitle.Paragraphs[index + 1];
-                if (next.StartTime.TotalMilliseconds - paragraph.EndTime.TotalMilliseconds > SetPeriodIfDistanceToNextIsMoreThan &&
-                    !paragraph.Text.EndsWith('.') &&
-                    !paragraph.Text.EndsWith('!') &&
-                    !paragraph.Text.EndsWith('?') &&
-                    !paragraph.Text.EndsWith(',') &&
-                    !paragraph.Text.EndsWith(':') &&
-                    !paragraph.Text.EndsWith(')') &&
-                    !paragraph.Text.EndsWith(']') &&
-                    !paragraph.Text.EndsWith('}'))
+
+                var distanceToNext = paragraph.EndTime.DistanceTo(next.StartTime);
+                
+                // line already terminated or too close to each other
+                if (distanceToNext <= SetPeriodIfDistanceToNextIsMoreThan || IsTerminated(paragraph.Text))
+                {
+                    continue;
+                }
+
+                if (AlwaysSetPeriodNoFurtherValidation(distanceToNext))
+                {
+                    paragraph.Text += ".";
+                }
+                else
                 {
-                    var gap = next.StartTime.TotalMilliseconds - paragraph.EndTime.TotalMilliseconds;
-                    if (gap > SetPeriodIfDistanceToNextIsMoreThanAlways)
+                    if (TwoLetterLanguageCode == "en" && (englishSkipLastWords.Contains(GetLastWord(paragraph.Text)) ||
+                                                          englishSkipFirstWords.Contains(GetFirstWord(next.Text))))
```

Comment:
Both `GetLastWord(paragraph.Text))` and  `GetFirstWord(next.Text)` are moved to where they are needed 
to avoid unnecessary method calls as only one of them needs to evaluated to true in order for the expression to be evaluated as true